pandas/_libs/tslib.pyx

```diff
@@ -126,10 +126,37 @@ cdef inline object create_datetime_from_ts(
     return datetime(dts.year, dts.month, dts.day, dts.hour,
                     dts.min, dts.sec, dts.us, tz)
 
+cdef inline object create_date_from_ts(
+        int64_t value, pandas_datetimestruct dts,
+        object tz, object freq):
+    """ convenience routine to construct a datetime.date from its parts """
+    return date(dts.year, dts.month, dts.day)
 
-def ints_to_pydatetime(ndarray[int64_t] arr, tz=None, freq=None, box=False):
-    # convert an i8 repr to an ndarray of datetimes or Timestamp (if box ==
-    # True)
+
+def ints_to_pydatetime(ndarray[int64_t] arr, tz=None, freq=None,
+                       box="datetime"):
+    """
+    Convert an i8 repr to an ndarray of datetimes, date or Timestamp
+
+    Parameters
+    ----------
+    arr  : array of i8 repr
+    tz   : the timezone to convert to,
```

Comment:
so doc-strings should be slightly different

```
arr : array of i8
tz : str, default None
    convert to this timezone
freq : str/Offset, default None
    freq to convert
box : {'datetime', 'timestamp', 'date'}, default 'datetime'
    # add what you have below here
```

Reply:
The asv results are:
`asv continuous -f 1.1 upstream/master issue18058 -b timeseries.DatetimeIndex.time_dti_time -b timeseries.DatetimeIndex.time_to_date -b timeseries.DatetimeIndex.time_to_pydatetime`

```
conda-py3.6-Cython-matplotlib-numexpr-numpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt
·· Installing into conda-py3.6-Cython-matplotlib-numexpr-numpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt
· Running 6 total benchmarks (2 commits * 1 environments * 3 benchmarks)
[  0.00%] · For pandas commit hash 04c90d90:
[  0.00%] ·· Building for conda-py3.6-Cython-matplotlib-numexpr-numpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt
[  0.00%] ·· Benchmarking conda-py3.6-Cython-matplotlib-numexpr-numpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt
[ 16.67%] ··· Running timeseries.DatetimeIndex.time_dti_time                                312ms
[ 33.33%] ··· Running timeseries.DatetimeIndex.time_to_date                                 40.6ms
[ 50.00%] ··· Running timeseries.DatetimeIndex.time_to_pydatetime                           46.9ms
[ 50.00%] · For pandas commit hash c176a3c2:
[ 50.00%] ·· Building for conda-py3.6-Cython-matplotlib-numexpr-numpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt
[ 50.00%] ·· Benchmarking conda-py3.6-Cython-matplotlib-numexpr-numpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt
[ 66.67%] ··· Running timeseries.DatetimeIndex.time_dti_time                               375ms
[ 83.33%] ··· Running timeseries.DatetimeIndex.time_to_date                                258±4ms
[100.00%] ··· Running timeseries.DatetimeIndex.time_to_pydatetime                          48.8±1ms       
      before           after          ratio
     [c176a3c2]       [04c90d90]
-           375ms            312ms     0.83  timeseries.DatetimeIndex.time_dti_time
-         258±4ms           40.6ms     0.16  timeseries.DatetimeIndex.time_to_date
```

Reply:
great!

we should fix time too :) (another PR)

Reply:
ah, yeah, i was about to ask